### ISSUE TITLE
[Bug] Model in building tab is blocking interface

### ISSUE THREAD

**Issue by GamyGamer** [KnuckleCracker/CW4-bug-tracker#436]
1. Screenshots help! 
 
![image](https://user-images.githubusercontent.com/49346764/101823139-a05c4980-3b2a-11eb-8cda-03bbf3c7b371.png)

2. Describe the bug 
   I have decided to add ERN maker as buildable unit, so I set build cost to 1000 (model hid one zero) and put it in "custom" tab but it will work on any tab. It also is hiding unit name and partialy Arg icon

**Comment by knucracker** [comment]
Can you reproduce this reliably? I jus tried but I get the expected results:
![image](https://user-images.githubusercontent.com/6733225/102415252-dcc5f480-3fc5-11eb-9d31-eb8edcf35e39.png)

If it something with your specific settings, you could export the cpack and then attach it here (as a zip).

**Comment by Regallion** [comment]
The specific issue can be checked in mission 136
![image](https://user-images.githubusercontent.com/40322714/102417947-d196b700-400d-11eb-8410-188ee03e74c4.png)
[BugMaker.zip](https://github.com/KnuckleCracker/CW4-bug-tracker/files/5705981/BugMaker.zip)

**Comment by Alophox** [comment]
The issue may be specific to units that are larger than standard in model size; footprint size doesn't do anything.
![image](https://user-images.githubusercontent.com/26010895/102430360-071caf80-3fd0-11eb-9b19-d3d97124a2fe.png)
Both the SILO and BERTHA are 5x5, for instance, and are not scaled differently, so they cover up parts of the name.
![image](https://user-images.githubusercontent.com/26010895/102430890-2c112280-3fd0-11eb-8f7e-6376b5a2f7ab.png)
The FLARE, by contrast, is 3x3 and looks normal.
![image](https://user-images.githubusercontent.com/26010895/102434102-52838d80-3fd1-11eb-97b5-212a01bf1225.png)
Making the FLARE ten times as large made it do this.

Custom units may not be scaled properly, while in-built ones are.

**Comment by knucracker** [comment]
This bug depended on the game resolution as well as the UI scale. The unit model was being scaled against an absolute scale not against whatever the actual button size was that it was going into. So it came out looking right at the resolution I test in within the unity dev environment. If the UI scale was changed a lot or the resolution was changed, the custom unit button image could get really large or really small relative to the button.

I've changed it to scale according to the actual button's  RectTransform width and height.
